src/main/java/hudson/plugins/git/extensions/impl/PruneStaleTag.java

```diff
@@ -0,0 +1,120 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+package hudson.plugins.git.extensions.impl;
+
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.GitSCMExtensionDescriptor;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.RemoteConfig;
+import org.eclipse.jgit.transport.URIish;
+import org.jenkinsci.plugins.gitclient.FetchCommand;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+
+/**
+ * Prune stale remote-tracking tags
+ *
+ * @author Nikolas Falco
+ * @since 4.2.1
+ */
+public class PruneStaleTag extends GitSCMExtension {
+    @DataBoundConstructor
+    public PruneStaleTag() {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void decorateFetchCommand(GitSCM scm,
+                                     @CheckForNull Run<?, ?> run,
+                                     GitClient git,
+                                     TaskListener listener,
+                                     FetchCommand cmd) throws IOException, InterruptedException, GitException {
+        listener.getLogger().println("Pruning obsolete local tags");
+
+        List<RemoteConfig> remoteRepos = run == null ? scm.getRepositories() : scm.getParamExpandedRepos(run, listener);
+        for (RemoteConfig remote : remoteRepos) {
+            for(URIish url : remote.getURIs()) {
+                FetchCommand fetch = git.fetch_().from(url, Arrays.asList(new RefSpec("+refs/tags/*:refs/tags/*")));
+                fetch.tags(false);
+                fetch.prune(true);
+                fetch.execute();
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        return o instanceof PruneStaleTag;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return PruneStaleTag.class.hashCode();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return "PruneStaleTag{}";
+    }
+
+    @Extension
+    public static class DescriptorImpl extends GitSCMExtensionDescriptor {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getDisplayName() {
+            return "Prune stale remote-tracking tags";
```

Comment:
A [stackoverflow response](https://stackoverflow.com/questions/15043450/git-branch-from-remote-tag) says that remote-tracking tags are not a recognized thing.  There are tags.  I think this should be rephrased.  You might consider: 
```suggestion
            return "Prune stale tags";
```